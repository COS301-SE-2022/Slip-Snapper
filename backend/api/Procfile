web: node apps/server.js