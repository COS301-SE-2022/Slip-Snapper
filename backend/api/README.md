# Commands 
    setup:
    > cd backend/api
    > yarn install

    To run Node server:
    > yarn start

    To test Node server:
    > yarn test