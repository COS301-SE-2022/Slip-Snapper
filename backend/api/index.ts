const express = require('express');
const fs = require('fs');

const app = express();

// const bodyParser = require('body-parser');
// app.use(bodyParser.json());

app.listen(1234, () =>{
    console.log('Server ready');
})

app.get('/example',async(req,res)=>{
    console.log(req.body);

    fs.readFile( __dirname + "/" + "database.json", 'utf8', function (err, data) {
        res.status(200).end(data);
    });
})