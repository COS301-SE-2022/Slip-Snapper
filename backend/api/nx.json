{
  "$schema": "./node_modules/nx/schemas/nx-schema.json",
  "npmScope": "slip-snapper",
  "affected": {
    "defaultBase": "master"
  },
  "cli": {
    "defaultCollection": "@nrwl/workspace"
  },
  "implicitDependencies": {
    "package.json": {
      "dependencies": "*",
      "devDependencies": "*"
    },
    ".eslintrc.json": "*"
  },
  "tasksRunnerOptions": {
    "default": {
      "runner": "@nrwl/nx-cloud",
      "options": {
        "cacheableOperations": [
          "build",
          "lint",
          "test",
          "e2e"
        ],
        "accessToken": "ZDQ4MzZjMDctMzVhNy00OWNkLWE3YmMtNWE5YTVkM2Q5MzE0fHJlYWQtd3JpdGU="
      }
    }
  },
  "targetDependencies": {
    "build": [
      {
        "target": "build",
        "projects": "dependencies"
      }
    ]
  }
}
