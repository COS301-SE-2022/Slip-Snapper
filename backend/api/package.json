{
  "name": "slip-snapper",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "start": "nodemon apps/server.js",
    "test": "jest src/app.test.js"
  },
  "private": true,
  "dependencies": {
    "cors": "^2.8.5",
    "express": "^4.18.1",
    "fs": "^0.0.1-security",
    "jest": "^28.1.0",
    "nodemon": "^2.0.16",
    "pdfkit": "^0.13.0",
    "supertest": "^6.2.3"
  },
  "devDependencies": {
    "@nrwl/cli": "14.1.5",
    "@nrwl/nx-cloud": "latest",
    "@nrwl/workspace": "14.1.5",
    "nx": "14.1.5",
    "prettier": "^2.5.1",
    "typescript": "~4.6.2"
  }
}
