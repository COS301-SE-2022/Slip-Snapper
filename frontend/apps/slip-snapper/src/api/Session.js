/**
 * Checks if session is still valid and if not,
 * logs the user out
 * 
 * @param {*} apiResponse Response from server
 * @returns to login page
 */
export async function destroySession(apiResponse) {
    if (apiResponse.data.message) {
        if (apiResponse.data.message === "Token has expired Login again to continue using the application") {
            const alert = document.createElement('ion-alert');
            alert.header = 'Session Expired';
            alert.subHeader = 'Your Session Has Expired';
            alert.message = 'Please login again.';
            alert.backdropDismiss = false;
            alert.buttons = [
                {
                    text: 'OK',
                    handler: () => {
                        localStorage.removeItem('user')
                        sessionStorage.removeItem('token')
                        window.location.replace("/login")
                        return;
                    }
                },
            ]

            document.body.appendChild(alert);
            await alert.present();
        }
        else {
            return;
        }
    } else {
        return;
    }
}