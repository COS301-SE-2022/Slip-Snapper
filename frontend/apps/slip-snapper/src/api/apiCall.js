const axios = require('axios');

/**
 * To process the text extracted by OCR
 * @param {*} ocr the extracted text
 * @returns the response as a promise
 */
export async function doProcessing(ocr){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'post',
        url:'http://localhost:55555/api/ocr/process',
        // data: {
        //   text: ocr,
        // }
        data: JSON.stringify({
            text: ocr,
        })
    })
}

export async function updateItemA(_item,userid,_name,_location,_quantity,_price,_type){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'post',
        url:'http://localhost:55555/api/item/update',
        data: JSON.stringify({
            itemid: _item,
            user: userid,
            name : _name,
            location : _location,
            quantity : _quantity,
            price : _price,
            type : _type
        })
    })
}

export async function getItemsA(userid){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url:'http://localhost:55555/api/item/all?userId='+userid
    })
}

export async function setBudgetA( userid, weekly, monthly){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'post',
        url: 'http://localhost:55555/api/report/budget',
        // data: JSON.stringify({
        //     userId: userid,
        //     weekly: weekly,
        //     monthly: monthly
        // })
        data: {
            userId: userid,
            weekly: weekly,
            monthly: monthly
        }
    })
}

export async function getBudgetA( userid ){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url: 'http://localhost:55555/api/report/budget?userId=1',
    })
}

export async function loginA( username, password){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'post',
        url: 'http://localhost:55555/api/user/login',
        body: JSON.stringify({
            username: username,
            password: password
        })
    })
}

export async function signupA( username, first, last, password){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'post',
        url: 'http://localhost:55555/api/user/signup',
        data: JSON.stringify({
          firstname: first,
          lastname: last,
          username: username,
          password: password,
        })
    })
}

export async function generateReportA( url ){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url: url,
      })
}

export async function getStatsA( userId ){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url: 'http://localhost:55555/api/report/statistics?userId=1',
      })
}

export async function addItemsA( userId, data, text){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'post',
        url: 'http://localhost:55555/api/item/add',
        body: JSON.stringify({
          userId: userId,
          location: data.text[1],
          date: data.text[0],
          total: data.text[4],
          data: text
        })
    })
}

export async function getAllUserReports(userId){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url: 'http://localhost:55555/api/report/user?userId=1',
      })
}

export async function getUserReport(userName, fileName){
    const url = 'http://localhost:55555/api/report/pdf?userName=' + userName + '&fileName=' + fileName
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url: url,
    })
}

export async function getRecentReports(userID){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'get',
        url: 'http://localhost:55555/api/report/recent?userId=1',
      })
}

//Temporary
export async function removeReport( userName, fileName , reportID){
    const headers = {
        "Content-Type": "application/json"
    }
    return axios({
        headers: headers,
        method: 'delete',
        url: 'http://localhost:55555/api/report/pdf',
        body: JSON.stringify({
          userName: userName,
          fileName: fileName,
          reportID: reportID
        })
    })
}
