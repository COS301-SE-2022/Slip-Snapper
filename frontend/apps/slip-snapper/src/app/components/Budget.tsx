import {
    IonAlert,
    IonInput,
    IonItem,
    IonProgressBar,
} from "@ionic/react";
import { useState } from "react";


function Budget() {
    const [food, setFood] = useState(false);
    const [fashion, setFashion] = useState(false);
    const [elec, setElec] = useState(false);
    const [houseHold, setHouseHold] = useState(false);
    const [other, setOther] = useState(false);

    const [amounts, setAmounts] = useState<string[]>(["100", "200", "300", "400", "500"]);

    return (
        <div>
            <IonItem onClick={() => setFood(true)} id="foodBudget" className="center-items" color="tertiary">
                <IonInput readonly value={"Food: R"+amounts[0]}></IonInput>
                <IonProgressBar id='foodBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={food}
                onDidDismiss={() => setFood(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food',placeholder: 'Insert Food Budget'},]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food,0);
                            isExceeded(parseFloat(alertData.food),"foodBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setFashion(true)} id="fashionBudget" className="center-items" color="tertiary">
                <IonInput readonly value={"Fashion: R" + amounts[1]}></IonInput>
                <IonProgressBar id='fashionBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={fashion}
                onDidDismiss={() => setFashion(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Fashion Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 1);
                            isExceeded(parseFloat(alertData.food), "fashionBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setElec(true)} id="electronicsBudget" className="center-items" color="tertiary">
                <IonInput readonly value={"Electronics: R" + amounts[2]}></IonInput>
                <IonProgressBar id='elecBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={elec}
                onDidDismiss={() => setElec(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Electronics Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 2);
                            isExceeded(parseFloat(alertData.food), "elecBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setHouseHold(true)} id="weekly-budget" className="center-items" color="tertiary">
                <IonInput readonly value={"Household: R" + amounts[3]}></IonInput>
                <IonProgressBar id='houseBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={houseHold}
                onDidDismiss={() => setHouseHold(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Fashion Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 3);
                            isExceeded(parseFloat(alertData.food), "houseBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setOther(true)} id="weekly-budget" className="center-items" color="tertiary">
                <IonInput readonly value={"Other: R" + amounts[4]}></IonInput>
                <IonProgressBar id='otherBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={other}
                onDidDismiss={() => setOther(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Fashion Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 4);
                            isExceeded(parseFloat(alertData.food), "otherBar")
                        }
                    }]}></IonAlert>
        </div>
    );

    function applyToBudget(newValue: string ,pos:number) {
        const val = parseFloat(newValue)

        if (!isNaN(val)) {
            amounts[pos]=val.toString()
            setAmounts(amounts)
        }
    }

  
    function isExceeded(budget:number,barID:string) {
        const totalFoodSpent = 88;
        const withinBudget = totalFoodSpent / budget

        if (totalFoodSpent >= budget && !isNaN(budget)) {
            document.getElementById(barID)?.setAttribute("color", "danger")
        }
        else if (totalFoodSpent >= budget / 2 && !isNaN(budget)) {
            document.getElementById(barID)?.setAttribute("color", "warning")
        }
        else if (!isNaN(budget)) {
            document.getElementById(barID)?.setAttribute("color", "success")
        }

        document.getElementById(barID)?.setAttribute("value", withinBudget.toString())

        if (budget === 0) {
            document.getElementById(barID)?.setAttribute("value", "0")
            document.getElementById(barID)?.setAttribute("color", "success")
        }

      
    }
}
export default Budget;