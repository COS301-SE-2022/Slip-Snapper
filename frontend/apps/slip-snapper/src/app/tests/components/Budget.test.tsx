import renderer from 'react-test-renderer';
import {
    IonAlert,
    IonFab,
    IonIcon,
    IonItem,
    IonProgressBar,
    IonText,
    useIonToast
} from "@ionic/react";
import { useEffect, useState } from "react";

import { create } from 'ionicons/icons'
import Filter7Icon from '@mui/icons-material/Filter7Outlined';
import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
import { getProfileData, setGeneralBudget } from "../../../api/apiCall"
import Budget, { updateBudgets } from "../../components/Budget";

jest.mock("../../api/apiCall");

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('<Budget>', () => {
  it('should render component', () => {
    expect(renderTree(<Budget 
    />).toJSON()).toMatchSnapshot();
  });
  
});


describe('updateBudgets', () => {
  it('should expose a function', () => {
		expect(updateBudgets).toBeDefined();
	});

});