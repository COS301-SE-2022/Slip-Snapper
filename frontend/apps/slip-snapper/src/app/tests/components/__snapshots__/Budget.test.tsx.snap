// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Budget> should render component 1`] = `
<div>
  <ion-item
    color="tertiary"
    id="foodBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Food: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="foodBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="fashionBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Fashion: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="fashionBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="electronicsBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Electronics: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="elecBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="houseBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Household: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="houseBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="hobbyBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Hobby: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="hobbyBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="healthcareBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Healthcare: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="healthcareBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="vehicleBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Vehicle: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="vehicleBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
  <ion-item
    color="tertiary"
    id="otherBudget"
    onClick={[Function]}
  >
    <ion-icon
      onClick={[Function]}
      src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Create</title><path d='M459.94 53.25a16.06 16.06 0 00-23.22-.56L424.35 65a8 8 0 000 11.31l11.34 11.32a8 8 0 0011.34 0l12.06-12c6.1-6.09 6.67-16.01.85-22.38zM399.34 90L218.82 270.2a9 9 0 00-2.31 3.93L208.16 299a3.91 3.91 0 004.86 4.86l24.85-8.35a9 9 0 003.93-2.31L422 112.66a9 9 0 000-12.66l-9.95-10a9 9 0 00-12.71 0z'/><path d='M386.34 193.66L264.45 315.79A41.08 41.08 0 01247.58 326l-25.9 8.67a35.92 35.92 0 01-44.33-44.33l8.67-25.9a41.08 41.08 0 0110.19-16.87l122.13-121.91a8 8 0 00-5.65-13.66H104a56 56 0 00-56 56v240a56 56 0 0056 56h240a56 56 0 0056-56V199.31a8 8 0 00-13.66-5.65z'/></svg>"
    />
    <ion-text>
      Other: R
      0.00
    </ion-text>
    <ion-progress-bar
      class="categoryProgressBar"
      id="otherBar"
      slot="end"
    />
    <br />
    <ion-fab
      edge={true}
      horizontal="end"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarMonthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
        />
      </svg>
    </ion-fab>
  </ion-item>
</div>
`;
